### ISSUE TITLE
Browse pages don't take into account sort configuration

### ISSUE THREAD

**Issue by ybnd** [DSpace/DSpace#8651]
**Describe the bug**

Browse page results aren't sorted as configured in REST

* The configured sort direction is not taken into account

* The sort type / order cannot be set for metadata indices, even though this should be possible [[according to the documentation](https://wiki.lyrasis.org/display/DSDOC7x/Configuration+Reference#ConfigurationReference-DefiningtheIndexes)](https://wiki.lyrasis.org/display/DSDOC7x/Configuration+Reference#ConfigurationReference-DefiningtheIndexes)

  > If you set a `<sort-type>` to be used, this sort type is not used on the values of the metadata fields but on the order of the items when listing all items that have a specific value of the metadata field.

* The default configuration sorts dates in ascending order, even though the documentation mentions that

  > descending may be useful for date-based indexes

**To Reproduce**

1. Run a local DSpace instance with the following browse configuration
   ```
   webui.browse.index.1 = dateissued:item:dateissued:desc
   webui.browse.index.2 = author:metadata:dc.contributor.*\,dc.creator:text:desc:dateissued
   ```
3. Open the "by issue date" browse page
   * Angular will send a request with `sort=default,ASC`
   * The response will be sorted by ascending date
4. Re-send that request without the `sort` parameter
   * The response will be sorted by descending date, as configured
5. Open the "by author" browse page
   * Angular will send a request with `sort=default,ASC`
   * The REST request will fail with HTTP 500, the following error will appear in the log
     ```
     2023-01-31 13:01:43,944 ERROR 2472f5f8-fc56-476e-9868-25549c647e08 755495a6-7d3f-43a8-9237-e3f25a13bb2b org.dspace.app.rest.exception.DSpaceApiExceptionControllerAdvice @ An exception has occurred (status:500)
     java.lang.IllegalStateException: The requested browse doesn't provide direct access to items you must specify a filter
     	at org.dspace.app.rest.repository.BrowseItemLinkRepository.listBrowseItems(BrowseItemLinkRepository.java:82) ~[classes/:?]
       ...
     ```

**Expected behavior**

* When no sort direction is specified, browse pages should sort results as configured in REST.
* When no sort direction is specified, Angular should update the "pagination component" to reflect the default sort direction as configured in REST.

**General thoughts**

While working on #8642 @YanaDePauw noted that browse could be replaced with discovery/Solr facets.
IMO this may be a more worthwhile time investment than maintaining it.

**Comment by tdonohue** [comment]
This ticket sounds like it may be related to #8646 .  As @mwoodiupui notes in that ticket, it appears that the `webui.itemlist.sort-option.*` settings in our dspace.cfg (which define the `<sort-type>` options available in browse) may not be fully functional in DSpace 7.

I haven't had a chance to dig into this at all, but I suspect we may have some "dead code" or "dead configs" here.  Things that used to work in 6.x, but no longer work in 7.x.   I agree that the approach here should either be to fix them, or possibly remove the dead code/configs to replace with Discovery/Solr configuration instead.

Moving this over to "Needs help" alongside that other ticket.   Volunteers are welcome if anyone is interested in teasing this apart & finding a solution.

**Comment by buithaihai** [comment]
DLCorp would like to claim this issue, along with #8646 and #7957 since it makes sense to tackle all of them simultaneously and publish in one PR. Our estimation is 24 hours of contribution for all three.

**Comment by tdonohue** [comment]
@buithaihai : Thanks. I'll assign this to you then.  Feel free to work on all 3 at once if it's easiest.

**Comment by tdonohue** [comment]
Reassigning to @jensvannerum as he's created a PR to fix this in DSpace/dspace-angular#3741

**Comment by JohnnyMendesC** [comment]
Greetings @tdonohue 

We at Neki-IT have encountered a similar issue to this one. When using the search on the community-page, collection-page and the search page, the sorting would mix up items, communities, and collections. 
We have developed a fix for this that ensures all items are displayed first, regardless of the current sorting order. Would this pull request be a good fit for this issue?

Looking forward to your feedback.
Best regards,
Johnny Mendes (Neki-IT)

**Comment by tdonohue** [comment]
Hi @JohnnyMendesC : This issue ticket already has a possible fix in a linked PR: https://github.com/DSpace/dspace-angular/pull/3741  So, as far as I'm aware, no additional help is necessary. The PR is just waiting on additional testing.

As for your proposed feature/change, it'd probably require a new ticket.  I'm not sure myself whether users would prefer that behavior or the current behavior.  It might be a question to ask to the [DSpace Community Advisory Team](https://wiki.lyrasis.org/display/cmtygp/DSpace+Community+Advisory+Team) once a ticket is created to describe the new example behavior (perhaps with screenshots or similar).

**Comment by Ma-Tador** [comment]
From what i understand, the https://github.com/DSpace/dspace-angular/pull/3741 will only fix the default sorting (ASC/DESC). But the sort option (dateissued) is still not considered. Will this also be solved soon?

**Comment by tdonohue** [comment]
@Ma-Tador : Is your comment meant for PR https://github.com/DSpace/dspace-angular/pull/3741?  It claims to solve this bug ticket.  But, if you are finding that it doesn't solve the bug, then I'd recommend adding your comment to that PR to help provide feedback there.

Regarding that PR, it looks like it's "stalled" waiting on the developer to address past feedback.  So, I'm uncertain of the status myself.

I'd love to also get this solved though, so if you are willing to help provide feedback on the PR, maybe that will prompt the developer to get back to that PR.  Essentially, this ticket is still waiting on a volunteer to build a "working" PR which we can test, approve and merge.
